test/conftest.py

```diff
@@ -17,9 +17,17 @@ def load_recipe_tests():
 
 
 def idfn(val):
-    return '{} {}'.format(val['recipe'], ' '.join(val['args'])).strip()
+    return '{} {}'.format(val['recipe'], ' '.join(val.get('args', []))).strip()
+
+
+def idinteg(val):
+    return val['recipe'].strip()
 
 
 def pytest_generate_tests(metafunc):
     if 'recipe_test' in metafunc.fixturenames:
-        metafunc.parametrize('recipe_test', list(load_recipe_tests()), ids=idfn)
+        metafunc.parametrize('recipe_test', list(load_recipe_tests('recipe_tests')), ids=idfn)
+    elif 'recipe_integration' in metafunc.fixturenames:
+        metafunc.parametrize('recipe_integration',
+                             list(load_recipe_tests("recipe_integration_tests")),
```

Comment:
This works, but since the inputs for these tests are *a lot* simpler than the recipe tests, I think we should just use basic pytest parametrization:
https://docs.pytest.org/en/latest/parametrize.html

Essentially we'll leave this `pytest_generate_tests` function alone and store the test input data in some kind of global data structure that will be fed into `pytest.mark.parametrize`. Here is an example of a test in mozilla-central that does something similar:
https://searchfox.org/mozilla-central/source/python/mozlint/test/test_pathutils.py#27

You can be creative with the data structure format (e.g you could have a dict with the name of the recipe to run as the keys).

Reply:
Thanks I updated the code. The pattern of test cases is:
```
TEST_CASES = [
    "recipe and parameter",
    ["put recipe to list to make it optional (xfail)"]
]
```